src/memory/test/tstmemory.cc

```diff
@@ -165,18 +162,22 @@ void tst_bad_alloc(rtt_dsxx::UnitTest &ut) {
     FAILMSG("Failed to catch a bad_alloc.");
   }
 
-#else // DRACO_DIAGNOSTICS & 2
-
-#endif // DRACO_DIAGNOSTICS & 2
-
   if (ut.numFails > num_fails_start)
     FAILMSG("Test failures in tst_bad_alloc.");
   else
-    PASSMSG("tst_bad_alloc completed sucessfully.");
+    PASSMSG("tst_bad_alloc completed successfully.");
 
   return;
 }
 
+//----------------------------------------------------------------------------------------//
+// Some compilers are clever enough to figure out that if you pass
+// std::numeric_limits<size_t>::max() to the new operator, you will always
+// blow away member, and so they will refuse to compile the code. We have
+// to use a bit of indirection to get such compilers to swallow the huge
+// allocation meant to deliberately blow away memory.
```

Comment:
Evidently, even this trick isn't enough.  

```
src/memory/test/tstmemory.cc:148:55: warning: argument 1 value ‘18446744073709551615’ exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
```

Can you try adding the following CPP macro to disable GCC's `-Walloc-size_larger-than` check?

```
#include <vector>

#if defined(__GNUC__) && !defined(__clang__)
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Walloc-size_larger-than"
#endif

<test code goes here>

#if defined(__GNUC__) && !defined(__clang__)
#pragma GCC diagnostic pop
#endif
```

Reply:
Done.